packages/babel-traverse/test/replacement.js

```diff
@@ -112,6 +112,33 @@ describe("path/replacement", function () {
       });
       expect(visitCounter).toBe(1);
     });
+
+    // https://github.com/babel/babel/issues/12386
+    it("updates pathCache with the replaced node", () => {
```

Comment:
I come up with a slightly simplified test from the regression example: https://github.com/babel/babel/pull/12391/commits/8d8782a701f2a1513d0ee186068780a5b2680658#diff-f6b6fd6b98ff2127cf882c020e42e695555b277d7073252187299fb7c3f875a2

This test does not depend on the behaviour of `proposal-optional-chaining` and `transform-typescript`.